File: simple/java/replace-conditional-with-polymorphism_after.java
abstract class Bird {
  ...
  abstract double getSpeed();
  ...
}

class European extends Bird {
  double getSpeed() {
    return getBaseSpeed();
  }
}
class African extends Bird {
  double getSpeed() {
    return getBaseSpeed() - getLoadFactor() * _numberOfCoconuts;
  }
}
class NorvegianBlue extends Bird {
  double getSpeed() {
    return (_isNailed) ? 0 : getBaseSpeed(_voltage);
  }
}

// Somewhere in Client code.
speed = bird.getSpeed();

File: simple/python/replace-conditional-with-polymorphism_after.py
class Bird:
    #...
    def getSpeed(self):
        pass

class European(Bird):

    def getSpeed(self):
        return self.getBaseSpeed()
    
    
class African(Bird):

    def getSpeed(self):
        return self.getBaseSpeed() - self.getLoadFactor() * self.numberOfCoconuts


class NorvegianBlue(Bird):

    def getSpeed():
        return 0 if self.isNailed else self.getBaseSpeed(self.voltage)

# Somewhere in client code
speed = bird.getSpeed()